tools/serial_listener.py

```diff
@@ -12,46 +12,56 @@
 exit handler
 """
 def exit_handler(signum, frame) :
-	print "\nexiting!\n"
-	s.close()
-	sys.exit()
+    print "\nexiting!\n"
+    s.close()
+    sys.exit()
+
+def waitKeyboard():
+    if (sys.platform == "win32") : #windows
+        CLEAR_CMD = "cls"
+    else :                 #some other OS (should insert more checks)
+        CLEAR_CMD = "clear"
+
+    while True:
+        raw_input()
+        os.system(CLEAR_CMD)
 
 
 #Main code####################################################################################
 def main(args):
-	if (sys.platform == "win32") : #windows
-		CLEAR_CMD = "cls"
-	else :				   #some other OS (should insert more checks)
-		CLEAR_CMD = "clear"
-		
-
-	
-	#initialize the serial port
-	s.port = args.port #this may change, depending on what port the OS gives the microcontroller
-	s.baudrate = args.baudrate #the baudrate may change in the future
-	s.open()		#attempt to open the serial port (there is no guard code, I'm assuming this does not fail)
-	
-	signal.signal(signal.SIGINT, exit_handler)
-	signal.signal(signal.SIGTERM, exit_handler) 
-	
-	
-	#clear the screen
-	os.system(CLEAR_CMD)
-	
-	#Main reading loop
-	while True:
-		print hex(ord(s.read(1)))
-		
+
+    #setup clearscreen thread
+    import threading
```

Comment:
I'm not very familiar with Python syntax and conventions, but shouldn't imports be done at the top of the file?